backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/CsvApi.java

```diff
@@ -50,10 +52,52 @@ private static String discardMetadata(Request request, Response response) {
   }
 
   static String mergeMetadata(Request request, Response response) {
-    SchemaMetadata schema = Emx2.fromRowList(getRowList(request));
-    getSchema(request).migrate(schema);
-    response.status(200);
-    return "add/update metadata success";
+    try {
+      SchemaMetadata schema = Emx2.fromRowList(getRowList(request));
+      getSchema(request).migrate(schema);
+      response.status(200);
+      return "add/update metadata success";
+    } catch (MolgenisException me) {
+      // the upload might actual table data instead of metadata
+      // but, we don't know which table because there is no filename (no multipart form)
+      // therefore, use findTableByColumns() to figure out where the data must go
+      Iterable<Row> rows = getRowList(request);
+      Row firstRow = rows.iterator().next();
+      Table t = findTableByColumns(firstRow.getColumnNames(), getSchema(request));
+      int count = t.save(getRowList(request));
+      response.status(200);
+      response.type(ACCEPT_CSV);
+      return "" + count;
+    }
+  }
+
+  /**
+   * Intersect a list of column names with the column names of tables of a specified schema. If we
+   * find exactly one table that contains all of these columns, that table is returned.
+   *
+   * @param checkColNames
+   * @param schema
+   * @return
+   */
+  static Table findTableByColumns(Set<String> checkColNames, Schema schema) {
+    List<Table> candidateTables = new ArrayList<>();
+    for (String tableName : schema.getTableNames()) {
+      Table t = schema.getTable(tableName);
+      List<String> intersect = new ArrayList<>();
+      for (String colName : t.getMetadata().getColumnNames()) {
+        if (checkColNames.contains(colName)) {
+          intersect.add(colName);
+        }
+      }
+      if (checkColNames.size() == intersect.size()) {
+        candidateTables.add(t);
+      }
+    }
```

Comment:
I feel like this could be changed to a lambda function to make it a bit more readable. 
Also, this part should be put into a function 'findCandidateTable()', that way it is more clear what this function actually does and what is preparation for doing so.